File: ewbapp/Models/Domain/PatrolChecklistItem.swift
import Foundation

struct PatrolChecklistItem: Codable, Identifiable {
    let id: UUID
    var label: String
    var isComplete: Bool
    var completedAt: Date?

    init(id: UUID = UUID(), label: String, isComplete: Bool = false, completedAt: Date? = nil) {
        self.id = id
        self.label = label
        self.isComplete = isComplete
        self.completedAt = completedAt
    }
}

File: app/src/main/java/org/yac/llamarangers/domain/model/PatrolChecklistItem.kt
package org.yac.llamarangers.domain.model

import kotlinx.serialization.Serializable

@Serializable
data class PatrolChecklistItem(
    val id: String = java.util.UUID.randomUUID().toString(),
    val label: String,
    val isCompleted: Boolean = false,
    val completedAt: Long? = null
)
